pkg/services/query/query.go

```diff
@@ -268,6 +273,31 @@ func (s *ServiceImpl) handleQuerySingleDatasource(ctx context.Context, user iden
 		req.Queries = append(req.Queries, q.query)
 	}
 
+	if s.features.IsEnabled(ctx, featuremgmt.FlagQueryMigrations) {
+		migReq := &backend.QueryMigrationRequest{
+			PluginContext: pCtx,
+			Queries:       req.Queries,
+		}
+
+		migrated, err := s.pluginClient.MigrateQuery(ctx, migReq)
```

Comment:
we REALLY need to find a process here that does not execute two round trips for every query.  

Or at least figure out why that is a bad idea -- not possible.

Reply:
IIUC, this could be managed/registered entirely in the SDK and executed

Reply:
> IIUC, this could be managed/registered entirely in the SDK and executed

That could be possible 🤔  I'll try that.